system-tests/test.py

```diff
@@ -179,16 +180,37 @@ def __init__(self,
                  fc_flags: str = None,
                  ld_flags: str = None):
         args: List[str] = []
+
         if fpp_flags:
-            args.append('--fpp-flags=' + fpp_flags)
-        if fc_flags:
-            args.append('--fc-flags=' + fc_flags)
-        if ld_flags:
-            args.append('--ld-flags=' + ld_flags)
+            # different config file name for fpp flag test
+            conf_file = test_directory/('stay_config.ini')
+        else:
+            conf_file = test_directory/'config.ini'
 
-        args.extend(['--exec-name', 'fab_test'])
-        args.append(target)
         args.append(str(test_directory))
+        args.append(str(conf_file))
+
+        config = configparser.ConfigParser(allow_no_value=True)
+        config['settings'] = {}
+        settings = config['settings']
+        settings['target'] = target
+        settings['exec-name'] = 'fab_test'
+        config['flags'] = {}
+        flags = config['flags']
+        if fpp_flags:
+            flags['fpp-flags'] = fpp_flags
+        else:
+            flags['fpp-flags'] = ''
+        if fc_flags:
+            flags['fc-flags'] = fc_flags
+        else:
+            flags['fc-flags'] = ''
+        if ld_flags:
+            flags['ld-flags'] = ld_flags
+        else:
+            flags['ld-flags'] = ''
+        with open(conf_file, 'w') as configfile:
+            config.write(configfile)
```

Comment:
Using the `configparser` library to build the test configuration file like this, while the correct course of action for application code, is probably not the best approach for test code. Tests tend to use static canned data wherever possible to avoid expanding the test surface. Every library included is effectively another moving part, another thing being tested.

In this case I would recommend a simple string and `.format()`.

Reply:
I can see your point and have changed how the test configuration is built.

Reply:
I can't see the change. Did you forget to "push" it?

Reply:
That's strange. I definitely pushed it, and can see it here on my end?